engine/classes/Elgg/Database.php

```diff
@@ -642,6 +642,12 @@ public function sanitizeInt($value, $signed = true) {
 	 * @return string
 	 */
 	public function sanitizeString($value) {
+
+		// use resource if established, but don't open a connection to do it.
```

Comment:
My thinking here is that I don't want to change the timing of DB connections being made in the boot process. This may be overly paranoid.